Review fixes by MM

manojmalik20 opened the pull request:
Memoized the getCustomStatus and getRecentCustomStatuses selector
Changed the name of parameters in position utils and modified handling topOffset in emoji picker tabs
Removed the dialogProps from the custom status modal oepening function everywhere
Removed the use of strong tags and used span with CSS
Added localization in the custom status modal and respective ids in i18n/en.json
Added placeholder in quick input component
Updated snapshot of Profile popover and tests of custom status selectors

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

#### Related Pull Requests
<!--
List all PRs related to resolving a ticket. For instance, if you submitted a PR to `mattermost/mattermost-server`, please include it here.
-->
- Has server changes (please link here)
- Has redux changes (please link here)
- Has mobile changes (please link here)

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.
-->

manojmalik20 committed:
Review fixes by MM
Memoized the getCustomStatus and getRecentCustomStatuses selector
Changed the name of parameters in position utils and modified handling topOffset in emoji picker tabs
Removed the dialogProps from the custom status modal oepening function everywhere
Removed the use of strong tags and used span with CSS
Added localization in the custom status modal and respective ids in i18n/en.json
Added placeholder in quick input component
Updated snapshot of Profile popover and tests of custom status selectors

chetanyakan commented on components/custom_status/custom_status_modal.tsx:
```diff
@@ -45,12 +45,17 @@ const defaultCustomStatusSuggestions: DefaultUserCustomStatus[] = [
     const [showEmojiPicker, setShowEmojiPicker] = useState<boolean>(false);
+    if (!currentCustomStatus) {
+        currentCustomStatus = {};
+    }
```
Should we merge this with the declaration on line 51?

chetanyakan commented on components/post_view/post_header/post_header_custom_status.tsx:
```diff
@@ -44,24 +45,24 @@ const PostHeaderCustomStatus = (props: ComponentProps) => {
-                </span>
-            </div>
-        );
+    if (!(customStatusEnabled && !isCustomStatusSet && showUpdateStatusButton && !isSystemMessage && isCurrentUserPost)) {
```
1. Please return null using a check: 
```js
if (!customStatusEnabled || isSystemMessage) {
    return null;
}
```
Do this before the if condition at line 31/32

2. Should we check `isCurrentUserPost` only using CSS?

chetanyakan commented on components/profile_popover/profile_popover.scss:
```diff
@@ -4,3 +4,7 @@
     padding: 0
 }
+
+.bold {
```
Don't create utility CSS class names.
```suggestion
.user-profile-popover__heading {
```

Also, update the usages.

manojmalik20 committed:
Fixed type-check issues and the width issue in the LHS channel name

chetanyakan commented on components/profile_popover/profile_popover.jsx:
```diff
@@ -291,11 +290,11 @@ class ProfilePopover extends React.PureComponent {
                     <button
                         className='user-popover__set-custom-status-btn'
-                        aria-label={Utils.localizeMessage('user_profile.custom-status.set-status', 'Set a status').toLowerCase()}
+                        aria-label={Utils.localizeMessage('user_profile.custom_status.set_status', 'Set a status').toLowerCase()}
```
This `aria-label` repeats the message already visible on the screen.
I think this is not needed as screen readers will read the other message and we can remove this.

Refer: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-label_attribute

***

On a separate note, should we use `aria-labelled-by` or some other similar tags?

chetanyakan commented on selectors/views/custom_status.ts:
```diff
@@ -3,34 +3,37 @@
-    const hasViewedCustomStatusModal = preferences[key] && preferences[key].value === Preferences.CUSTOM_STATUS_MODAL_VIEWED;
-    return !hasViewedCustomStatusModal;
+    const hasViewedCustomStatusModal = get(state, Preferences.CATEGORY_CUSTOM_STATUS, Preferences.NAME_CUSTOM_STATUS_TUTORIAL_STATE);
+    return hasViewedCustomStatusModal !== Preferences.CUSTOM_STATUS_MODAL_VIEWED;
```
This looks really suspicious. maybe we need to tweak some variable names here.

manojmalik20 replied:
This was suggested in the PR comments.

chetanyakan replied:
What does the `hasViewedCustomStatusModal` variable name mean according to you and how is it used here?

manojmalik20 replied:
I have renamed this variable to customStatusTutorialState

manojmalik20 committed:
Fixed some issues and bugs
Added remaining localization ids in en.json
Modified showing update status button logic only via CSS and fixed the bug of header not showing on comments
Changed names of some classes like bold
Changed names of some variables
